crates/starknet-server/src/api/http/models.rs

```diff
@@ -16,18 +18,46 @@ pub(crate) struct LoadPath {
 #[derive(Deserialize, Debug)]
 pub(crate) struct PostmanLoadL1MessagingContract {
     #[serde(rename = "networkUrl")]
-    network_url: String,
-    address: ContractAddress,
+    pub network_url: String,
+    pub address: Option<String>,
+
+    // TODO: THIS IS FOR TESTING PURPOSE ONLY.
```

Comment:
Important point to remove when some configuration can be available from the API.

Reply:
Is this a reminder to yourself to remove this before merging? What do you mean by "configuration available from the API"?

Reply:
I was referring to this comment found in the code: https://github.com/0xSpaceShard/starknet-devnet-rs/blob/c8af0bbb85b71ad644f4c0f1e040f4b68713d859/crates/starknet-server/src/api/mod.rs#L15.

Does this make sense? Or I'm not in the right and I should read the private key from an already know location?

Reply:
the private key of what?

Reply:
In order to deploy the messaging contract on L1 node and send transaction to mock the messages on the `MockStarknetMessaging` contract, I do need a private key for the EOA account.

Reply:
I mean, if it's only for usage with HardHat, Anvil etc... the private key does not worth that much and can totally be passed in the RPC body along with the other arguments.

Reply:
Here's what we did in Devnet-py: https://github.com/0xSpaceShard/starknet-devnet/blob/e477aa1bbe2348ba92af2a69c32d2eef2579d863/starknet_devnet/postman_wrapper.py#L158

Is this relevant? IIRC, we were relying on the L1 network to have at least one unlocked account

Reply:
From the python documentation here https://web3py.readthedocs.io/en/stable/web3.eth.account.html#local-vs-hosted-keys it looks like they are pre-defining the keys for the corresponding node in the code.

In the case of `ethers`, I think they are agnostic of the local node used for testing.

So we can totally replicate what `web3` in python is doing, by pre-registering in the messaging struct some accounts, for `HardHat` and `Anvil` at least.
But this means, we've to update the code anytime we want to support an other tool.

1. It's not a problem and we start by having this list somewhere in the code.
2. We can do both, to ensure a good evolution of Devnet without the need to recompile. If the user provides a `private_key` in the `postman_load` endpoint, it's fine we use it. If no `private_key` is given (like in devnet-py), we attempt to deploy with the first default account of `HardHat`/`Anvil`/...

WDYT?

Reply:
I like the suggestion in 2. We might also need to pass the account address together with the private key?

Reply:
Yes sure the address is always passed like the devnet py API. We just have this extension with the private key that is optional.
If the address passed is known as pre-registered, we use a private key we have internally.
If the address is not known we expected a private key along the network url. We can then send bad request or more appropriate status.

Reply:
Sounds good

Reply:
After testing a bit, this commit 9dfe0c980457bb20179d8ede8bd2552666191dc2 introduces the behavior very similar to devnet-py.
If no `private_key` passed in the model of `postman/load_l1_messaging_contract`, we use the `accounts[0]` of the default wallet for eth devnets (at least hardhat and anvil compatible).

And to avoid problem with new eth devnet or other configuration of eth devnet the user may have, we accept (as an extension from devnet-py) a `private_key` (which is optional).
If this `private_key` is given, it is used.

We don't need the account address as the only computation we need is signing transaction, where only the private key is required.

Good for you @FabijanC?

Reply:
1. Any reason why you are not defaulting to `ETH_ACCOUNT_DEFAULT.private_key` using `.unwrap_or`?
2. But more importantly, I don't see why we now need this dummy `ETH_ACCOUNT_DEFAULT` when it wasn't needed in devnet-py; there we just relied on `web3.eth.accounts` (likely relying on the [eth_accounts RPC method](https://docs.infura.io/networks/ethereum/json-rpc-methods/eth_accounts))  having the account at index 0 unlocked.

Reply:
Oh.. I didn't know about this RPC method.
But this method is only returning the address of the account for what I've just tested, not it's private key. This leads to me second point below.

About unlocking account I don't know (yet) how to do it I've never used it, and the `ethers` methods are asking for a signer. I'll then investigate how this can be done.

If you have any knowledge on this, I'll be happy to learn about it.

Reply:
> Any reason why you are not defaulting to ETH_ACCOUNT_DEFAULT.private_key using .unwrap_or?

nope, good catch. I had more code here before, didn't refacto. Thanks for the remark.

Reply:
@FabijanC to follow up on this, do you want to remove `private_key` and we give to the user the expected accounts? (which are already setup by default for anvil and hardhat).

Reply:
Fixed in 1341be6b69005e33eb62c9c534802b787b6802ed.